migration/incremental/015-tracing-compression.sql

```diff
@@ -234,6 +241,167 @@ GRANT USAGE ON TYPE ps_trace.trace_id TO prom_reader;
 -- ps_trace.trace_id and uuid are binary coercible
 CREATE CAST(ps_trace.trace_id as uuid) WITHOUT FUNCTION AS IMPLICIT;
 CREATE CAST(uuid AS ps_trace.trace_id) WITHOUT FUNCTION AS IMPLICIT;
+/* Move the old tag_map type out of the way */
+ALTER DOMAIN ps_trace.tag_map
+    RENAME TO tag_map_old;
+ALTER DOMAIN ps_trace.tag_map_old
+    SET SCHEMA _ps_trace;
+
+/* Define custom tag_map and tag_v json aliases
+ * NOTE: We need to keep an eye on upstream jsonb type
+ * and keep these up-to-date.
+ */
+
+CREATE TYPE ps_trace.tag_map;
+CREATE TYPE _ps_trace.tag_v;
+
+CREATE OR REPLACE FUNCTION ps_trace.tag_map_in(cstring)
+RETURNS ps_trace.tag_map
+LANGUAGE internal
+IMMUTABLE PARALLEL SAFE STRICT
+AS $function$jsonb_in$function$
+;
+
+CREATE OR REPLACE FUNCTION ps_trace.tag_map_out(ps_trace.tag_map)
+RETURNS cstring
+LANGUAGE internal
+IMMUTABLE PARALLEL SAFE STRICT
+AS $function$jsonb_out$function$
+;
+
+CREATE OR REPLACE FUNCTION ps_trace.tag_map_send(ps_trace.tag_map)
+RETURNS bytea
+LANGUAGE internal
+IMMUTABLE PARALLEL SAFE STRICT
+AS $function$jsonb_send$function$
+;
+
+CREATE OR REPLACE FUNCTION ps_trace.tag_map_recv(internal)
+RETURNS ps_trace.tag_map
+LANGUAGE internal
+IMMUTABLE PARALLEL SAFE STRICT
+AS $function$jsonb_recv$function$
+;
+
+DO
+$do$
+
+/* Create subscript_handler function for pg v14+ and the type accordingly.
+ * For pg v13 jsonb type doesn't have a subscript_handler function so it shall
+ * be omitted.
+ */
+DECLARE
+    _pg_version int4 := current_setting('server_version_num')::int4;
+BEGIN
+    IF (_pg_version >= 140000) THEN
+        EXECUTE
+            'CREATE OR REPLACE FUNCTION ps_trace.tag_map_subscript_handler(internal) ' ||
+                'RETURNS internal '                 ||
+                'LANGUAGE internal '                ||
+                'IMMUTABLE PARALLEL SAFE STRICT '   ||
```

Comment:
does this mean that, if I install promscale on v13, and then upgrade to v14, I will never get the subscript functionality?

Reply:
Yes, which shall be fixed if we address Mat's comment on adding all these to idempotent branch.